Test/Unit/Block/JsTest.php

```diff
@@ -169,18 +179,39 @@ public function testGetConnectJsUrlForProductionMode()
     /**
      * @inheritdoc
      */
-    public function testGetCheckoutKey()
+    public function testGetCheckoutKey_checkoutPage()
     {
-        $storeId = 0;
-        $key = 'pKv_pOzRTEST.TESTkEIjTEST.TEST01f0d15501cd7548c1953f6666b2689f2e5a20198c5d7f886c004913TEST';
+        $key = 'pKv_pOzRTEST.TESTkEIjTEST.TEST01f0d15501cd7548c1953f6666b2689f2e5a20198c5d7f886c004913TESTCHECKOUTPAGE';
+
         $this->configHelper->expects($this->any())
-            ->method('getAnyPublishableKey')
+            ->method('getPublishableKeyPayment')
             ->will($this->returnValue($key));
+         $this->requestMock->expects($this->any())
+             ->method('getFullActionName')
+             ->will($this->returnValue(HelperConfig::CHECKOUT_PAGE_ACTION));
 
         $result = $this->block->getCheckoutKey();
+        $this->assertStringStartsWith('pKv_', $result, '"Publishable Key Payment Only" doesn\'t work properly');
+        $this->assertEquals(strlen($key), strlen($result), '"Publishable Key Payment Only" has an invalid length');
+    }
 
-        $this->assertStringStartsWith('pKv_', $result, '"Any Publishable Key" not working properly');
-        $this->assertEquals(strlen($key), strlen($result), '"Any Publishable Key" have invalid length');
+    /**
+     * @inheritdoc
+     */
+    public function testGetCheckoutKey_shoppingCartPage()
```

Comment:
I suggest to add a `@dataProvider someName` to simplify test cases. Also it would be more simply in future to add new scenarious especially negative scenarious (e.x when we do not have a publishable key here)